activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -16,13 +16,55 @@ def self.find(version)
       V6_0 = Current
 
       class V5_2 < V6_0
+        module TableDefinition
+          def timestamps(**options)
+            options[:precision] ||= nil
+            super
+          end
+        end
+
         module CommandRecorder
           def invert_transaction(args, &block)
             [:transaction, args, block]
           end
         end
 
+        def create_table(table_name, **options)
+          if block_given?
+            super { |t| yield compatible_table_definition(t) }
+          else
+            super
+          end
+        end
+
+        def change_table(table_name, **options)
+          if block_given?
+            super { |t| yield compatible_table_definition(t) }
+          else
+            super
+          end
```

Comment:
Why do we need these similar overrides?

Reply:
It is to inject `V5_2::TableDefinition` to the table definition instance.

https://github.com/rails/rails/pull/34970/files#diff-2a8be25f82da6b3935cc6a41300a1b01R19`

If the `change_table` is removed, the `test_timestamps_doesnt_set_precision_on_change_table` is failed.

https://github.com/rails/rails/pull/34970/files#diff-ec048630132cce280a9445022318906eR140